src/constraints/outage_constraints.jl

```diff
@@ -302,23 +312,20 @@ function add_MG_storage_dispatch_constraints(m,p)
         )
     end
     
-    # Dispatch to MG electrical storage is no greater than inverter capacity
-    # and can't charge the battery unless binMGStorageUsed = 1
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
-    )
-    
-    # Dispatch from MG storage is no greater than inverter capacity
-    # and can't discharge from storage unless binMGStorageUsed = 1
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
-    )
-    
-    # Dispatch to and from electrical storage is no greater than power capacity
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
-            + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
-    )
+    # Dispatch to and from MG electrical storage is no greater than power capacity
+    if "ElectricStorage" in p.s.storage.types.dc_coupled
+        @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+            m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts] / p.s.storage.attr["ElectricStorage"].inverter_efficiency_fraction
+                + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.ac_coupled_with_storage) * p.s.storage.attr["ElectricStorage"].rectifier_efficiency_fraction
+                + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.dc_coupled_with_storage)
+        )
+    else
+        @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+            m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
+                + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.ac_coupled_with_storage)
+                + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.dc_coupled_with_storage) * p.s.storage.attr["ElectricStorage"].inverter_efficiency_fraction #TODO: remove term or leave for future flexibility?
```

Comment:
If "ElectricStorage" in **not** in `p.s.storage.types.dc_coupled`, why would we need this here?

Reply:
This is flow from dc coupled techs out inverter to ac coupled storage, which is currently is not allowed. With multiple storage and adding a bool input to say whether this flow is allowed, this term would be relevant, which is why I wrote that TODO.